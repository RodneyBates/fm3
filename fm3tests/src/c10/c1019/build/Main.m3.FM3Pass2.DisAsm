 ItkEOF()
 ItkLeftEnd()
 ItkModuleLt(L1(16_1),[1,0])
 ItkBlockDeclListLt(L1(16_1),[9,2])
 ItkProcWBodyLt(Id8"P")
 ItkDeclNo(D14,[9,12])
 ItkSignatureProperLt([9,14])
 ItkFormalsListLt(L0(16_0),[9,14])
 ItkFormalsListRt(L0(16_0),[9,14])
 ItkResultTypeAbsent([11,2])
 ItkRaisesSetLt(L0(16_0),[11,2])
 ItkRaisesSetRt(L0(16_0),[11,2])
 ItkSignatureProperRt([9,14])
 ItkProcBodyLt([11,2])
 ItkBlockDeclListLt(L7(16_7),[11,4])
 ItkVarDeclLt([11,8])
 ItkVarDeclIdListLt(L1(16_1),[11,8])
 ItkDeclNo(D7,[11,8])
 ItkVarDeclIdListRt(L1(16_1),[11,8])
 ItkVarDeclType([11,10])
 ItkReservedId(Rid"INTEGER",[11,12])
 ItkVarDeclVal([14,2])
 ItkDeclValAbsent()
 ItkVarDeclRt([11,8])
 ItkBlockDeclListSep(L1(16_1),[14,2])
 ItkConstDeclLt([14,10])
 ItkDeclNo(D1,[14,10])
 ItkConstDeclColon([14,14],FALSE)
 ItkDeclTypeAbsent()
 ItkConstDeclEquals([14,14])
 ItkIntLit(16_0(0),[14,16])
 ItkConstDeclRt([14,10])
 ItkBlockDeclListSep(L2(16_2),[15,2])
 ItkConstDeclLt([15,10])
 ItkDeclNo(D2,[15,10])
 ItkConstDeclColon([15,14],FALSE)
 ItkDeclTypeAbsent()
 ItkConstDeclEquals([15,14])
 ItkIntLit(16_1(1),[15,16])
 ItkConstDeclRt([15,10])
 ItkBlockDeclListSep(L3(16_3),[16,2])
 ItkConstDeclLt([16,10])
 ItkDeclNo(D3,[16,10])
 ItkConstDeclColon([16,13],FALSE)
 ItkDeclTypeAbsent()
 ItkConstDeclEquals([16,13])
 ItkIntLit(16_2(2),[16,15])
 ItkConstDeclRt([16,10])
 ItkBlockDeclListSep(L4(16_4),[18,2])
 ItkConstDeclLt([18,10])
 ItkDeclNo(D4,[18,10])
 ItkConstDeclColon([18,13],FALSE)
 ItkDeclTypeAbsent()
 ItkConstDeclEquals([18,13])
 ItkIntLit(16_3(3),[18,15])
 ItkConstDeclRt([18,10])
 ItkBlockDeclListSep(L5(16_5),[19,2])
 ItkConstDeclLt([19,10])
 ItkDeclNo(D5,[19,10])
 ItkConstDeclColon([19,14],FALSE)
 ItkDeclTypeAbsent()
 ItkConstDeclEquals([19,14])
 ItkIntLit(16_4(4),[19,16])
 ItkConstDeclRt([19,10])
 ItkBlockDeclListSep(L6(16_6),[20,2])
 ItkConstDeclLt([20,10])
 ItkDeclNo(D6,[20,10])
 ItkConstDeclColon([20,13],FALSE)
 ItkDeclTypeAbsent()
 ItkConstDeclEquals([20,13])
 ItkIntLit(16_5(5),[20,15])
 ItkConstDeclRt([20,10])
 ItkBlockDeclListRt(L7(16_7),[11,4])
 ItkStmtListLt(L5(16_5),[24,6])
 ItkBecomesLt([24,8])
 ItkIdRefDeclNo(D7,[24,6])
 ItkBecomesInfix([24,8])
 ItkIdRefDeclNo(D1,[24,11])
 ItkBecomesRt([24,8])
 ItkStmtListSep(L0(16_0),[25,4])
 ItkBecomesLt([25,8])
 ItkIdRefDeclNo(D7,[25,6])
 ItkBecomesInfix([25,8])
 ItkIdRefDeclNo(D2,[25,11])
 ItkBecomesRt([25,8])
 ItkStmtListSep(L1(16_1),[26,4])
 ItkBecomesLt([26,8])
 ItkIdRefDeclNo(D7,[26,6])
 ItkBecomesInfix([26,8])
 ItkIdRefDeclNo(D3,[26,11])
 ItkBecomesRt([26,8])
 ItkStmtListSep(L2(16_2),[28,4])
 ItkBecomesLt([28,8])
 ItkIdRefDeclNo(D7,[28,6])
 ItkBecomesInfix([28,8])
 ItkIdRefDeclNo(D4,[28,11])
 ItkBecomesRt([28,8])
 ItkStmtListSep(L3(16_3),[29,4])
 ItkBecomesLt([29,8])
 ItkIdRefDeclNo(D7,[29,6])
 ItkBecomesInfix([29,8])
 ItkIdRefDeclNo(D5,[29,11])
 ItkBecomesRt([29,8])
 ItkStmtListSep(L4(16_4),[30,4])
 ItkBecomesLt([30,8])
 ItkIdRefDeclNo(D7,[30,6])
 ItkBecomesInfix([30,8])
 ItkIdRefDeclNo(D6,[30,11])
 ItkBecomesRt([30,8])
 ItkStmtListRt(L5(16_5),[24,6])
 ItkProcBodyRt([11,2])
 ItkProcWBodyRt(Id8"P")
 ItkBlockDeclListRt(L1(16_1),[9,2])
 ItkModuleBegin(L1(16_1),[33,2])
 ItkStmtListLt(L0(16_0),[34,2])
 ItkStmtListRt(L0(16_0),[34,2])
 ItkModuleRt(L1(16_1),[1,0])
 ItkRightEnd()
 ItkBOF()
     0
