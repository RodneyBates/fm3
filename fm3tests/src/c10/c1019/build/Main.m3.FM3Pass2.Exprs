From unit expression map, Expr No 0 NIL

From unit expression map, Expr No 1 16_****************
 ExpStackLink = NIL
 ExpType = NIL
 ExpRefConstVal = NIL
 ExpScalarConstVal = 0
 ExpLoTypeInfoRef = 
 ExpReachedDeclNoSet = {}
 ExpSelfExprNo = 1
 ExpPosition = [30,11]
 ExpOpcode = <RidNull>
 ExpDownKind = EkNull
 ExpUpKind = EkNull
 ExpKind = EkIdentRef
 ExpState = EsUnresolved
 ExpIsConst = FALSE
 ExpConstValIsKnown = FALSE
 ExpIsUsable = TRUE
 ExpIsLegalRecursive = FALSE
 ExpIsDesignator = FALSE
 ExpIsWritable = FALSE
 ExpIsPresent = TRUE
 ExpRefTypeIsUntraced = FALSE
 ExpArrayTypeIsOpen = FALSE
 (* From subtype ExprRemoteRefTyp: *)
 ExpRemoteUnitNo = 3
 ExpRemoteDeclNo = 1


From unit expression map, Expr No 2 16_****************
 ExpStackLink = ExprNo 1 at 16_**************** [30,11]
 ExpType = NIL
 ExpRefConstVal = NIL
 ExpScalarConstVal = 0
 ExpLoTypeInfoRef = 
 ExpReachedDeclNoSet = {}
 ExpSelfExprNo = 2
 ExpPosition = [30,6]
 ExpOpcode = <RidNull>
 ExpDownKind = EkNull
 ExpUpKind = EkNull
 ExpKind = EkIdentRef
 ExpState = EsUnresolved
 ExpIsConst = FALSE
 ExpConstValIsKnown = FALSE
 ExpIsUsable = TRUE
 ExpIsLegalRecursive = FALSE
 ExpIsDesignator = FALSE
 ExpIsWritable = FALSE
 ExpIsPresent = TRUE
 ExpRefTypeIsUntraced = FALSE
 ExpArrayTypeIsOpen = FALSE
 (* From subtype ExprIdentRefTyp: *)
 ExpIdentDeclNo = 6


From unit expression map, Expr No 3 16_****************
 ExpStackLink = ExprNo 0 at 16_**************** [29,11]
 ExpType = NIL
 ExpRefConstVal = NIL
 ExpScalarConstVal = 0
 ExpLoTypeInfoRef = 
 ExpReachedDeclNoSet = {}
 ExpSelfExprNo = 3
 ExpPosition = [29,6]
 ExpOpcode = <RidNull>
 ExpDownKind = EkNull
 ExpUpKind = EkNull
 ExpKind = EkIdentRef
 ExpState = EsUnresolved
 ExpIsConst = FALSE
 ExpConstValIsKnown = FALSE
 ExpIsUsable = TRUE
 ExpIsLegalRecursive = FALSE
 ExpIsDesignator = FALSE
 ExpIsWritable = FALSE
 ExpIsPresent = TRUE
 ExpRefTypeIsUntraced = FALSE
 ExpArrayTypeIsOpen = FALSE
 (* From subtype ExprIdentRefTyp: *)
 ExpIdentDeclNo = 6


From unit expression map, Expr No 4 16_****************
 ExpStackLink = ExprNo 0 at 16_**************** [28,11]
 ExpType = NIL
 ExpRefConstVal = NIL
 ExpScalarConstVal = 0
 ExpLoTypeInfoRef = 
 ExpReachedDeclNoSet = {}
 ExpSelfExprNo = 4
 ExpPosition = [28,6]
 ExpOpcode = <RidNull>
 ExpDownKind = EkNull
 ExpUpKind = EkNull
 ExpKind = EkIdentRef
 ExpState = EsUnresolved
 ExpIsConst = FALSE
 ExpConstValIsKnown = FALSE
 ExpIsUsable = TRUE
 ExpIsLegalRecursive = FALSE
 ExpIsDesignator = FALSE
 ExpIsWritable = FALSE
 ExpIsPresent = TRUE
 ExpRefTypeIsUntraced = FALSE
 ExpArrayTypeIsOpen = FALSE
 (* From subtype ExprIdentRefTyp: *)
 ExpIdentDeclNo = 6


From unit expression map, Expr No 5 16_****************
 ExpStackLink = ExprNo 4 at 16_**************** [28,6]
 ExpType = NIL
 ExpRefConstVal = NIL
 ExpScalarConstVal = 0
 ExpLoTypeInfoRef = 
 ExpReachedDeclNoSet = {}
 ExpSelfExprNo = 5
 ExpPosition = [26,11]
 ExpOpcode = <RidNull>
 ExpDownKind = EkNull
 ExpUpKind = EkNull
 ExpKind = EkIdentRef
 ExpState = EsUnresolved
 ExpIsConst = FALSE
 ExpConstValIsKnown = FALSE
 ExpIsUsable = TRUE
 ExpIsLegalRecursive = FALSE
 ExpIsDesignator = FALSE
 ExpIsWritable = FALSE
 ExpIsPresent = TRUE
 ExpRefTypeIsUntraced = FALSE
 ExpArrayTypeIsOpen = FALSE
 (* From subtype ExprIdentRefTyp: *)
 ExpIdentDeclNo = 9


From unit expression map, Expr No 6 16_****************
 ExpStackLink = ExprNo 5 at 16_**************** [26,11]
 ExpType = NIL
 ExpRefConstVal = NIL
 ExpScalarConstVal = 0
 ExpLoTypeInfoRef = 
 ExpReachedDeclNoSet = {}
 ExpSelfExprNo = 6
 ExpPosition = [26,6]
 ExpOpcode = <RidNull>
 ExpDownKind = EkNull
 ExpUpKind = EkNull
 ExpKind = EkIdentRef
 ExpState = EsUnresolved
 ExpIsConst = FALSE
 ExpConstValIsKnown = FALSE
 ExpIsUsable = TRUE
 ExpIsLegalRecursive = FALSE
 ExpIsDesignator = FALSE
 ExpIsWritable = FALSE
 ExpIsPresent = TRUE
 ExpRefTypeIsUntraced = FALSE
 ExpArrayTypeIsOpen = FALSE
 (* From subtype ExprIdentRefTyp: *)
 ExpIdentDeclNo = 6


From unit expression map, Expr No 7 16_****************
 ExpStackLink = ExprNo 6 at 16_**************** [26,6]
 ExpType = NIL
 ExpRefConstVal = NIL
 ExpScalarConstVal = 0
 ExpLoTypeInfoRef = 
 ExpReachedDeclNoSet = {}
 ExpSelfExprNo = 7
 ExpPosition = [25,11]
 ExpOpcode = <RidNull>
 ExpDownKind = EkNull
 ExpUpKind = EkNull
 ExpKind = EkIdentRef
 ExpState = EsUnresolved
 ExpIsConst = FALSE
 ExpConstValIsKnown = FALSE
 ExpIsUsable = TRUE
 ExpIsLegalRecursive = FALSE
 ExpIsDesignator = FALSE
 ExpIsWritable = FALSE
 ExpIsPresent = TRUE
 ExpRefTypeIsUntraced = FALSE
 ExpArrayTypeIsOpen = FALSE
 (* From subtype ExprIdentRefTyp: *)
 ExpIdentDeclNo = 8


From unit expression map, Expr No 8 16_****************
 ExpStackLink = ExprNo 7 at 16_**************** [25,11]
 ExpType = NIL
 ExpRefConstVal = NIL
 ExpScalarConstVal = 0
 ExpLoTypeInfoRef = 
 ExpReachedDeclNoSet = {}
 ExpSelfExprNo = 8
 ExpPosition = [25,6]
 ExpOpcode = <RidNull>
 ExpDownKind = EkNull
 ExpUpKind = EkNull
 ExpKind = EkIdentRef
 ExpState = EsUnresolved
 ExpIsConst = FALSE
 ExpConstValIsKnown = FALSE
 ExpIsUsable = TRUE
 ExpIsLegalRecursive = FALSE
 ExpIsDesignator = FALSE
 ExpIsWritable = FALSE
 ExpIsPresent = TRUE
 ExpRefTypeIsUntraced = FALSE
 ExpArrayTypeIsOpen = FALSE
 (* From subtype ExprIdentRefTyp: *)
 ExpIdentDeclNo = 6


From unit expression map, Expr No 9 16_****************
 ExpStackLink = ExprNo 8 at 16_**************** [25,6]
 ExpType = NIL
 ExpRefConstVal = NIL
 ExpScalarConstVal = 0
 ExpLoTypeInfoRef = 
 ExpReachedDeclNoSet = {}
 ExpSelfExprNo = 9
 ExpPosition = [24,11]
 ExpOpcode = <RidNull>
 ExpDownKind = EkNull
 ExpUpKind = EkNull
 ExpKind = EkIdentRef
 ExpState = EsUnresolved
 ExpIsConst = FALSE
 ExpConstValIsKnown = FALSE
 ExpIsUsable = TRUE
 ExpIsLegalRecursive = FALSE
 ExpIsDesignator = FALSE
 ExpIsWritable = FALSE
 ExpIsPresent = TRUE
 ExpRefTypeIsUntraced = FALSE
 ExpArrayTypeIsOpen = FALSE
 (* From subtype ExprIdentRefTyp: *)
 ExpIdentDeclNo = 7


From unit expression map, Expr No 10 16_****************
 ExpStackLink = ExprNo 9 at 16_**************** [24,11]
 ExpType = NIL
 ExpRefConstVal = NIL
 ExpScalarConstVal = 0
 ExpLoTypeInfoRef = 
 ExpReachedDeclNoSet = {}
 ExpSelfExprNo = 10
 ExpPosition = [24,6]
 ExpOpcode = <RidNull>
 ExpDownKind = EkNull
 ExpUpKind = EkNull
 ExpKind = EkIdentRef
 ExpState = EsUnresolved
 ExpIsConst = FALSE
 ExpConstValIsKnown = FALSE
 ExpIsUsable = TRUE
 ExpIsLegalRecursive = FALSE
 ExpIsDesignator = FALSE
 ExpIsWritable = FALSE
 ExpIsPresent = TRUE
 ExpRefTypeIsUntraced = FALSE
 ExpArrayTypeIsOpen = FALSE
 (* From subtype ExprIdentRefTyp: *)
 ExpIdentDeclNo = 6


From unit expression map, Expr No 11 16_****************
 ExpStackLink = ExprNo 10 at 16_**************** [24,6]
 ExpType = NIL
 ExpRefConstVal = NIL
 ExpScalarConstVal = 2
 ExpLoTypeInfoRef = 
 ExpReachedDeclNoSet = {}
 ExpSelfExprNo = 11
 ExpPosition = [16,15]
 ExpOpcode = StkIntLit
 ExpDownKind = EkNull
 ExpUpKind = EkValue
 ExpKind = EkLiteral
 ExpState = EsResolved 
 ExpIsConst = TRUE
 ExpConstValIsKnown = TRUE
 ExpIsUsable = TRUE
 ExpIsLegalRecursive = TRUE
 ExpIsDesignator = FALSE
 ExpIsWritable = FALSE
 ExpIsPresent = TRUE
 ExpRefTypeIsUntraced = FALSE
 ExpArrayTypeIsOpen = FALSE
 (* From subtype ExprConstValueTyp: *)


From unit expression map, Expr No 12 16_****************
 ExpStackLink = ExprNo 10 at 16_**************** [24,6]
 ExpType = NIL
 ExpRefConstVal = NIL
 ExpScalarConstVal = 1
 ExpLoTypeInfoRef = 
 ExpReachedDeclNoSet = {}
 ExpSelfExprNo = 12
 ExpPosition = [15,16]
 ExpOpcode = StkIntLit
 ExpDownKind = EkNull
 ExpUpKind = EkValue
 ExpKind = EkLiteral
 ExpState = EsResolved 
 ExpIsConst = TRUE
 ExpConstValIsKnown = TRUE
 ExpIsUsable = TRUE
 ExpIsLegalRecursive = TRUE
 ExpIsDesignator = FALSE
 ExpIsWritable = FALSE
 ExpIsPresent = TRUE
 ExpRefTypeIsUntraced = FALSE
 ExpArrayTypeIsOpen = FALSE
 (* From subtype ExprConstValueTyp: *)


From unit expression map, Expr No 13 16_****************
 ExpStackLink = ExprNo 10 at 16_**************** [24,6]
 ExpType = NIL
 ExpRefConstVal = NIL
 ExpScalarConstVal = 0
 ExpLoTypeInfoRef = 
 ExpReachedDeclNoSet = {}
 ExpSelfExprNo = 13
 ExpPosition = [14,16]
 ExpOpcode = StkIntLit
 ExpDownKind = EkNull
 ExpUpKind = EkValue
 ExpKind = EkLiteral
 ExpState = EsResolved 
 ExpIsConst = TRUE
 ExpConstValIsKnown = TRUE
 ExpIsUsable = TRUE
 ExpIsLegalRecursive = TRUE
 ExpIsDesignator = FALSE
 ExpIsWritable = FALSE
 ExpIsPresent = TRUE
 ExpRefTypeIsUntraced = FALSE
 ExpArrayTypeIsOpen = FALSE
 (* From subtype ExprConstValueTyp: *)


From unit expression map, Expr No 14 16_****************
 ExpStackLink = ExprNo 10 at 16_**************** [24,6]
 ExpType = NIL
 ExpRefConstVal = NIL
 ExpScalarConstVal = 0
 ExpLoTypeInfoRef = 
 ExpReachedDeclNoSet = {}
 ExpSelfExprNo = 14
 ExpPosition = [11,12]
 ExpOpcode = INTEGER
 ExpDownKind = EkNull
 ExpUpKind = EkNull
 ExpKind = EkNull
 ExpState = EsUnresolved
 ExpIsConst = FALSE
 ExpConstValIsKnown = FALSE
 ExpIsUsable = TRUE
 ExpIsLegalRecursive = TRUE
 ExpIsDesignator = FALSE
 ExpIsWritable = FALSE
 ExpIsPresent = TRUE
 ExpRefTypeIsUntraced = FALSE
 ExpArrayTypeIsOpen = FALSE
 (* From subtype ExprReservedIdRefTyp: *)


