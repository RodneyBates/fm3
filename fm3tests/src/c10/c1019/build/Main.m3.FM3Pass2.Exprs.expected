From unit expression map, Expr No 0 NIL

From unit expression map, Expr No 1 16_****************
 ExpStackLink = ExprNo 0 at 16_**************** [30,11]
 ExpType = NIL
 ExpRefConstVal = NIL
 ExpScalarConstVal = 2
 ExpLoTypeInfoRef = 
 ExpReachedDeclNoSet = {}
 ExpSelfExprNo = 1
 ExpPosition = [16,15]
 ExpOpcode = <RidNull>
 ExpDownKind = <null>
 ExpUpKind = value
 ExpKind = <null>
 ExpState = EsResolved 
 ExpIsConst = TRUE
 ExpConstValIsKnown = TRUE
 ExpIsUsable = TRUE
 ExpIsLegalRecursive = TRUE
 ExpIsDesignator = FALSE
 ExpIsWritable = FALSE
 ExpIsPresent = TRUE
 ExpRefTypeIsUntraced = FALSE
 (* From subtype ExprConstValueTyp: *)


From unit expression map, Expr No 2 16_****************
 ExpStackLink = ExprNo 0 at 16_**************** [30,11]
 ExpType = NIL
 ExpRefConstVal = NIL
 ExpScalarConstVal = 1
 ExpLoTypeInfoRef = 
 ExpReachedDeclNoSet = {}
 ExpSelfExprNo = 2
 ExpPosition = [15,16]
 ExpOpcode = <RidNull>
 ExpDownKind = <null>
 ExpUpKind = value
 ExpKind = <null>
 ExpState = EsResolved 
 ExpIsConst = TRUE
 ExpConstValIsKnown = TRUE
 ExpIsUsable = TRUE
 ExpIsLegalRecursive = TRUE
 ExpIsDesignator = FALSE
 ExpIsWritable = FALSE
 ExpIsPresent = TRUE
 ExpRefTypeIsUntraced = FALSE
 (* From subtype ExprConstValueTyp: *)


From unit expression map, Expr No 3 16_****************
 ExpStackLink = ExprNo 0 at 16_**************** [30,11]
 ExpType = NIL
 ExpRefConstVal = NIL
 ExpScalarConstVal = 0
 ExpLoTypeInfoRef = 
 ExpReachedDeclNoSet = {}
 ExpSelfExprNo = 3
 ExpPosition = [14,16]
 ExpOpcode = <RidNull>
 ExpDownKind = <null>
 ExpUpKind = value
 ExpKind = <null>
 ExpState = EsResolved 
 ExpIsConst = TRUE
 ExpConstValIsKnown = TRUE
 ExpIsUsable = TRUE
 ExpIsLegalRecursive = TRUE
 ExpIsDesignator = FALSE
 ExpIsWritable = FALSE
 ExpIsPresent = TRUE
 ExpRefTypeIsUntraced = FALSE
 (* From subtype ExprConstValueTyp: *)


From unit expression map, Expr No 4 16_****************
 ExpStackLink = ExprNo 0 at 16_**************** [30,11]
 ExpType = NIL
 ExpRefConstVal = NIL
 ExpScalarConstVal = 0
 ExpLoTypeInfoRef = 
 ExpReachedDeclNoSet = {}
 ExpSelfExprNo = 4
 ExpPosition = [11,12]
 ExpOpcode = INTEGER
 ExpDownKind = <null>
 ExpUpKind = <null>
 ExpKind = <null>
 ExpState = EsUnresolved
 ExpIsConst = FALSE
 ExpConstValIsKnown = FALSE
 ExpIsUsable = TRUE
 ExpIsLegalRecursive = TRUE
 ExpIsDesignator = FALSE
 ExpIsWritable = FALSE
 ExpIsPresent = TRUE
 ExpRefTypeIsUntraced = FALSE
 (* From subtype ExprReservedIdRefTyp: *)


